RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -118,16 +116,16 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const ChamberHitContainer& u
     }
 
     //change the tresholds if it's time to look for displaced mu segments
-    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
-      doCollisions = false;
+    if(l_doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
```

Comment:
I can't imagine it can be responsible of any non-reproducibility.

Anyhow this line is potentially wrong: "rechits.size()" is unsigned, therefore the difference can be unpredictable if used_rh>rechits.size() (can it happen?). 

It should be better rewritten as either
- int(rechits.size())-used_rh>2
- int(rechits.size())>used_rh+2